// @forge/util/packages/ari/chunk-6HAR4ROV.mjs
import {
  CommerceTransactionAccountAriResourceOwner,
  CommerceTransactionAccountAriResourceType
} from "./chunk-IG6GXX55.mjs";
import {
  RegisteredAri
} from "./chunk-PA4Y7TWI.mjs";
import {
  AriParser
} from "./chunk-2UUYVBKT.mjs";

// src/commerce/transaction-account/manifest.ts
var commerceTransactionAccountAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: CommerceTransactionAccountAriResourceOwner,
  resourceType: CommerceTransactionAccountAriResourceType,
  resourceIdSlug: "{txaId}",
  resourceIdSegmentFormats: {
    txaId: /[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}/
    // eslint-disable-line no-useless-escape
  }
};

// src/commerce/transaction-account/index.ts
var CommerceTransactionAccountAri = class _CommerceTransactionAccountAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._txaId = opts.resourceIdSegmentValues.txaId;
  }
  get txaId() {
    return this._txaId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: commerceTransactionAccountAriStaticOpts.qualifier,
      platformQualifier: commerceTransactionAccountAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: commerceTransactionAccountAriStaticOpts.resourceOwner,
      resourceType: commerceTransactionAccountAriStaticOpts.resourceType,
      resourceId: `${opts.txaId}`,
      resourceIdSegmentValues: {
        txaId: opts.txaId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, commerceTransactionAccountAriStaticOpts);
    return new _CommerceTransactionAccountAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, commerceTransactionAccountAriStaticOpts);
    return new _CommerceTransactionAccountAri(opts);
  }
  getVariables() {
    return {
      txaId: this.txaId
    };
  }
};

export {
  CommerceTransactionAccountAri
};
