// @forge/util/packages/ari/chunk-RS2IWKXH.mjs
import {
  JiraOperationsComponentAriResourceOwner,
  JiraOperationsComponentAriResourceType
} from "./chunk-PHMOFVGN.mjs";
import {
  RegisteredAri
} from "./chunk-PA4Y7TWI.mjs";
import {
  AriParser
} from "./chunk-2UUYVBKT.mjs";

// src/jira/operations-component/manifest.ts
var jiraOperationsComponentAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^[a-zA-Z0-9-]+$"),
  resourceOwner: JiraOperationsComponentAriResourceOwner,
  resourceType: JiraOperationsComponentAriResourceType,
  resourceIdSlug: "activation/{activationId}/{providerAppId}/{operationsComponentId}",
  resourceIdSegmentFormats: {
    activationId: /[a-zA-Z0-9\-]+/,
    // eslint-disable-line no-useless-escape
    providerAppId: /[a-zA-Z0-9-._]+/,
    // eslint-disable-line no-useless-escape
    operationsComponentId: /[a-zA-Z0-9\-_.~@:{}=]+(\/[a-zA-Z0-9\-_.~@:{}=]+)*/
    // eslint-disable-line no-useless-escape
  }
};

// src/jira/operations-component/index.ts
var JiraOperationsComponentAri = class _JiraOperationsComponentAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._siteId = opts.cloudId || "", this._activationId = opts.resourceIdSegmentValues.activationId, this._providerAppId = opts.resourceIdSegmentValues.providerAppId, this._operationsComponentId = opts.resourceIdSegmentValues.operationsComponentId;
  }
  get siteId() {
    return this._siteId;
  }
  get activationId() {
    return this._activationId;
  }
  get providerAppId() {
    return this._providerAppId;
  }
  get operationsComponentId() {
    return this._operationsComponentId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: jiraOperationsComponentAriStaticOpts.qualifier,
      platformQualifier: jiraOperationsComponentAriStaticOpts.platformQualifier,
      cloudId: opts.siteId,
      resourceOwner: jiraOperationsComponentAriStaticOpts.resourceOwner,
      resourceType: jiraOperationsComponentAriStaticOpts.resourceType,
      resourceId: `activation/${opts.activationId}/${opts.providerAppId}/${opts.operationsComponentId}`,
      resourceIdSegmentValues: {
        activationId: opts.activationId,
        providerAppId: opts.providerAppId,
        operationsComponentId: opts.operationsComponentId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, jiraOperationsComponentAriStaticOpts);
    return new _JiraOperationsComponentAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, jiraOperationsComponentAriStaticOpts);
    return new _JiraOperationsComponentAri(opts);
  }
  getVariables() {
    return {
      siteId: this.siteId,
      activationId: this.activationId,
      providerAppId: this.providerAppId,
      operationsComponentId: this.operationsComponentId
    };
  }
};

export {
  JiraOperationsComponentAri
};
