// @forge/util/packages/ari/chunk-OH62MPVY.mjs
import {
  GoogleSpreadsheetAriResourceOwner,
  GoogleSpreadsheetAriResourceType
} from "./chunk-466Z3YAU.mjs";
import {
  RegisteredAri
} from "./chunk-PA4Y7TWI.mjs";
import {
  AriParser
} from "./chunk-2UUYVBKT.mjs";

// src/google/spreadsheet/manifest.ts
var googleSpreadsheetAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: GoogleSpreadsheetAriResourceOwner,
  resourceType: GoogleSpreadsheetAriResourceType,
  resourceIdSlug: "{docId}",
  resourceIdSegmentFormats: {
    docId: /[a-zA-Z0-9_-]+/
    // eslint-disable-line no-useless-escape
  }
};

// src/google/spreadsheet/index.ts
var GoogleSpreadsheetAri = class _GoogleSpreadsheetAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._docId = opts.resourceIdSegmentValues.docId;
  }
  get docId() {
    return this._docId;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: googleSpreadsheetAriStaticOpts.qualifier,
      platformQualifier: googleSpreadsheetAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: googleSpreadsheetAriStaticOpts.resourceOwner,
      resourceType: googleSpreadsheetAriStaticOpts.resourceType,
      resourceId: `${opts.docId}`,
      resourceIdSegmentValues: {
        docId: opts.docId
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, googleSpreadsheetAriStaticOpts);
    return new _GoogleSpreadsheetAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, googleSpreadsheetAriStaticOpts);
    return new _GoogleSpreadsheetAri(opts);
  }
  getVariables() {
    return {
      docId: this.docId
    };
  }
};

export {
  GoogleSpreadsheetAri
};
