// @forge/util/packages/ari/index.mjs
import "./chunk-WRVV7GZO.mjs";
import {
  VirtualAgentIntentQuestionProjectionAri
} from "./chunk-H3VPAB3T.mjs";
import "./chunk-FRZCZ4T6.mjs";
import {
  VirtualAgentIntentRuleProjectionAri
} from "./chunk-XFVP3BRG.mjs";
import "./chunk-MJ72T4ZB.mjs";
import {
  VirtualAgentIntentTemplateAri
} from "./chunk-FACB27PD.mjs";
import "./chunk-66RZPSB4.mjs";
import {
  VirtualAgentConfigurationAri
} from "./chunk-BHOK6O4L.mjs";
import "./chunk-TPOGUUFE.mjs";
import {
  VirtualAgentFlowEditorAri
} from "./chunk-RITOXI47.mjs";
import "./chunk-ETXO2665.mjs";
import {
  VirtualAgentIntentProjectionAri
} from "./chunk-KMVPOHAE.mjs";
import "./chunk-MPJ7GQCW.mjs";
import "./chunk-ATN7NKOC.mjs";
import {
  UnifiedHelpWorkspaceAri
} from "./chunk-RNKQKQA7.mjs";
import "./chunk-GXNUL5AQ.mjs";
import {
  UnifiedHelpRoleAri
} from "./chunk-Y2GMHCDM.mjs";
import "./chunk-BQV74DFA.mjs";
import {
  UnifiedHelpSiteAri
} from "./chunk-5SEALRQ7.mjs";
import "./chunk-OKKAA5RV.mjs";
import "./chunk-4MWZBX5W.mjs";
import {
  TrelloWorkspaceAri
} from "./chunk-ZPFQVODV.mjs";
import "./chunk-KVD2HPFJ.mjs";
import {
  TrelloListAri
} from "./chunk-7OKXSENE.mjs";
import "./chunk-TVSKQH7F.mjs";
import {
  TrelloRoleAri
} from "./chunk-XUNT5L3I.mjs";
import "./chunk-I5GECIAC.mjs";
import {
  TrelloSiteAri
} from "./chunk-FZCR5GBW.mjs";
import "./chunk-P3FZABZE.mjs";
import {
  TrelloUserAri
} from "./chunk-EUOBS3S7.mjs";
import "./chunk-OZFY5Y6K.mjs";
import {
  TrelloBoardAri
} from "./chunk-LV5I2AUV.mjs";
import "./chunk-HI4UEG2F.mjs";
import {
  TrelloCardAri
} from "./chunk-GYB2JTKI.mjs";
import "./chunk-L5R7QT5M.mjs";
import {
  TrelloEnterpriseAri
} from "./chunk-GEX46XWJ.mjs";
import "./chunk-A3QGWDCS.mjs";
import "./chunk-D3TDYNKN.mjs";
import {
  TownsquareTagAri
} from "./chunk-4ASWZ6G6.mjs";
import "./chunk-7QVDZS4D.mjs";
import {
  TownsquareProjectAri
} from "./chunk-2A5PTN2K.mjs";
import "./chunk-ICBJW6N7.mjs";
import {
  TownsquareQuestionAri
} from "./chunk-BMHAMW43.mjs";
import "./chunk-IKIQRIWE.mjs";
import {
  TownsquareRoleAri
} from "./chunk-65GMXC2K.mjs";
import "./chunk-RRNPSY4J.mjs";
import {
  TownsquareSiteAri
} from "./chunk-VLO2J5Y6.mjs";
import "./chunk-MC7VKTYC.mjs";
import {
  TownsquareGoalAri
} from "./chunk-ZAOTSBFG.mjs";
import "./chunk-FPGL2PN7.mjs";
import {
  TownsquareHelpPointerAri
} from "./chunk-ADBNHMVA.mjs";
import "./chunk-GROHW3KB.mjs";
import {
  TownsquareLearningAri
} from "./chunk-ORH5L6OW.mjs";
import "./chunk-4J5O222V.mjs";
import "./chunk-LERUK3EA.mjs";
import {
  SupportCustomerAri
} from "./chunk-BAS6M5UR.mjs";
import "./chunk-PHBXUJAW.mjs";
import "./chunk-7BIQUOJP.mjs";
import {
  TeamsTeamAri
} from "./chunk-GHAQCLJY.mjs";
import "./chunk-MUICIWTK.mjs";
import {
  TownsquareCommentAri
} from "./chunk-4EFKXCDP.mjs";
import "./chunk-GVAPRDNZ.mjs";
import "./chunk-2EFBVLUL.mjs";
import {
  SearchEventAri
} from "./chunk-4UU46D7V.mjs";
import "./chunk-MLTMULJ4.mjs";
import "./chunk-TYQPBEKA.mjs";
import {
  StatuspageRoleAri
} from "./chunk-U74IDWBR.mjs";
import "./chunk-BT2GYN5W.mjs";
import {
  StatuspageSiteAri
} from "./chunk-UR55BUGU.mjs";
import "./chunk-6IR7P2SS.mjs";
import "./chunk-NTJ25HLA.mjs";
import {
  RuntimeAuthClientRoleAri
} from "./chunk-TKH6CCRT.mjs";
import "./chunk-N3DYIPHN.mjs";
import "./chunk-MGHXHAL5.mjs";
import {
  PlatformServicesStreamhubSchemaAri
} from "./chunk-UMWI7QXS.mjs";
import "./chunk-KM3YDOZJ.mjs";
import "./chunk-EHBNQJ47.mjs";
import {
  PlatformOrgAri
} from "./chunk-NVDJASQK.mjs";
import "./chunk-ON6PRZUL.mjs";
import {
  PlatformSecureTunnelAri
} from "./chunk-DRV2ZYTW.mjs";
import "./chunk-7AEMSYL6.mjs";
import {
  PlatformSiteAri
} from "./chunk-WG2XWYSC.mjs";
import "./chunk-QGU7KDYS.mjs";
import {
  PlatformLifecycleResourcePackageTypeAri
} from "./chunk-ZTWFSSSP.mjs";
import "./chunk-YUROJJWW.mjs";
import {
  PlatformLifecycleResourcePackageAri
} from "./chunk-PS3C2TAU.mjs";
import "./chunk-AUDUMQTZ.mjs";
import {
  PlatformLifecycleResourceAri
} from "./chunk-W3B3Y4FO.mjs";
import "./chunk-T36754GS.mjs";
import {
  PlatformOrgUserAri
} from "./chunk-J7CQFIR2.mjs";
import "./chunk-REH6WMJF.mjs";
import {
  PlatformClassificationTagAri
} from "./chunk-LP2XJ5J2.mjs";
import "./chunk-JOF4WRFT.mjs";
import "./chunk-2K5F5KXA.mjs";
import {
  PassionfruitUserAri
} from "./chunk-CUG2SMHT.mjs";
import "./chunk-Q2W2OS63.mjs";
import "./chunk-WL6TPSYP.mjs";
import {
  PeoplePerftoolPerformanceAri
} from "./chunk-WTXSCS6P.mjs";
import "./chunk-BBPB2HIH.mjs";
import {
  PeoplePerftoolCalibrationAri
} from "./chunk-GGJRQUJU.mjs";
import "./chunk-YFDO5OIQ.mjs";
import {
  PeoplePerftoolFeedbackAri
} from "./chunk-BDHBO5CO.mjs";
import "./chunk-BUB6ILHZ.mjs";
import "./chunk-P6ODAFPN.mjs";
import {
  PapiApiAri
} from "./chunk-GWT4UNXI.mjs";
import "./chunk-5IDHPJZF.mjs";
import {
  PapiRoleAri
} from "./chunk-ZM3GPI7D.mjs";
import "./chunk-NLJJ6GJ4.mjs";
import "./chunk-UW7FK2QY.mjs";
import {
  OpsgenieWorkspaceAri
} from "./chunk-UY2MEVZC.mjs";
import "./chunk-Y2SQLTAK.mjs";
import {
  OpsgenieScheduleAri
} from "./chunk-OFQY7N2T.mjs";
import "./chunk-OFBJZXGI.mjs";
import {
  OpsgenieSiteAri
} from "./chunk-RJHR4WH5.mjs";
import "./chunk-3A7NUHDU.mjs";
import {
  OpsgenieTeamAri
} from "./chunk-QNAAHRUK.mjs";
import "./chunk-5GDLDNUH.mjs";
import {
  OpsgenieTimelineAri
} from "./chunk-DVQF75RB.mjs";
import "./chunk-T4JHTCBL.mjs";
import {
  OpsgenieIntegrationAri
} from "./chunk-6IRGDEA2.mjs";
import "./chunk-KNRSSKUB.mjs";
import {
  OpsgenieNotificationAri
} from "./chunk-VOGU5CWY.mjs";
import "./chunk-WBXUDE5Y.mjs";
import {
  OpsgenieRoleAri
} from "./chunk-HVJJ22MX.mjs";
import "./chunk-7ZPWMXWE.mjs";
import {
  OpsgenieScheduleRotationAri
} from "./chunk-MSLIULJD.mjs";
import "./chunk-ILPXSBCV.mjs";
import {
  OpsgenieIncidentAlertLinkAri
} from "./chunk-TTP2F7T2.mjs";
import "./chunk-JCLOAQNZ.mjs";
import {
  OpsgenieIncidentStatusUpdateAri
} from "./chunk-SQP5CMSO.mjs";
import "./chunk-RO4YS42G.mjs";
import {
  OpsgenieIncidentAri
} from "./chunk-QBUGZDEX.mjs";
import "./chunk-TLNXXINJ.mjs";
import {
  OpsgenieIncomingCallHistoryAri
} from "./chunk-IMBFQDTD.mjs";
import "./chunk-W4G5P22J.mjs";
import {
  OpsgenieDeploymentServiceLinkAri
} from "./chunk-GOVJSWIB.mjs";
import "./chunk-VYEUWCEP.mjs";
import {
  OpsgenieDeploymentAri
} from "./chunk-XSIBGA2X.mjs";
import "./chunk-BVX743QJ.mjs";
import {
  OpsgenieEscalationAri
} from "./chunk-W3TCFCU4.mjs";
import "./chunk-C3Q7ZVPL.mjs";
import {
  OpsgenieEventAri
} from "./chunk-EONGG2SD.mjs";
import "./chunk-PW5GVXRG.mjs";
import {
  OpsgenieAlertAri
} from "./chunk-YNGBANCC.mjs";
import "./chunk-PZKFI7JM.mjs";
import {
  OpsgenieApiRequestMetricAri
} from "./chunk-44ER4CCS.mjs";
import "./chunk-R4SGT2SX.mjs";
import {
  OpsgenieCallRoutingAri
} from "./chunk-TXNPCYZ5.mjs";
import "./chunk-P2OBBGSQ.mjs";
import {
  OpsgenieCustomRoleAri
} from "./chunk-4VI4FOQT.mjs";
import "./chunk-RW54AUFT.mjs";
import "./chunk-IXJTRB7L.mjs";
import {
  OpsgenieAccountLoginAri
} from "./chunk-YSKELIVG.mjs";
import "./chunk-B2UP53JZ.mjs";
import {
  OpsgenieAccountSettingsAri
} from "./chunk-DB67C5D5.mjs";
import "./chunk-OZDNDKVC.mjs";
import {
  OpsgenieAlertRecipientLinkAri
} from "./chunk-QBOS25EO.mjs";
import "./chunk-JXI5TFG4.mjs";
import {
  OauthClientAri
} from "./chunk-46BDSDTZ.mjs";
import "./chunk-SLRY2ZFP.mjs";
import "./chunk-DU3KZ7XN.mjs";
import {
  MediaFileAri
} from "./chunk-SOOKC52U.mjs";
import "./chunk-D3IDOZFP.mjs";
import "./chunk-SV3OSS6S.mjs";
import {
  MercuryProgramAri
} from "./chunk-BPZ52OPF.mjs";
import "./chunk-YNUAOINR.mjs";
import {
  MercuryRoleAri
} from "./chunk-E3EIKCY3.mjs";
import "./chunk-2CIDR5KT.mjs";
import {
  MercurySiteAri
} from "./chunk-CRBTDEBI.mjs";
import "./chunk-SEACGWCF.mjs";
import {
  MercuryCommentAri
} from "./chunk-EXQVDUWK.mjs";
import "./chunk-IP7NNLNW.mjs";
import {
  MercuryProgramStatusUpdateAri
} from "./chunk-MFJQT7EB.mjs";
import "./chunk-DXGTR4X7.mjs";
import "./chunk-ITJPM4K6.mjs";
import "./chunk-ZNQVUOOQ.mjs";
import {
  MeasurementEmailUuidAri
} from "./chunk-I36GCRAQ.mjs";
import "./chunk-TZWB442U.mjs";
import {
  MeasurementSiteUserAri
} from "./chunk-VX4V753Y.mjs";
import "./chunk-CHIUNEWO.mjs";
import {
  MeasurementUserAri
} from "./chunk-OHRVWON3.mjs";
import "./chunk-G4CB4RQJ.mjs";
import "./chunk-Y2UB4MCB.mjs";
import {
  LinkingPlatformDatasourceAri
} from "./chunk-LMC6BNTB.mjs";
import "./chunk-ARLPVCER.mjs";
import "./chunk-ZM3JBOHH.mjs";
import {
  MakerSpaceGroupAri
} from "./chunk-TJSD5JZK.mjs";
import "./chunk-TI7RQAMT.mjs";
import {
  MarketingCustomerDomainAri
} from "./chunk-YXVHFCOC.mjs";
import "./chunk-VAIM2B7I.mjs";
import "./chunk-V7XF5L23.mjs";
import {
  JiraWorkflowSchemeAri
} from "./chunk-B6QVGI5Q.mjs";
import "./chunk-GF5WTUWI.mjs";
import {
  JiraWorkflowAri
} from "./chunk-RCK3VP2W.mjs";
import "./chunk-EYJJDCXK.mjs";
import {
  JiraWorklogAri
} from "./chunk-YKT47GF3.mjs";
import "./chunk-V5CUEKO2.mjs";
import {
  JiraWorkspaceAri
} from "./chunk-LYNMFCQN.mjs";
import "./chunk-KOJWXOCJ.mjs";
import {
  JiraVersionApproverAri
} from "./chunk-2S7LNTW5.mjs";
import "./chunk-CLH2DEO3.mjs";
import {
  JiraVersionAri
} from "./chunk-DWJNWRBQ.mjs";
import "./chunk-LN36NVHR.mjs";
import {
  JiraVulnerabilityHistoryAri
} from "./chunk-KIC3TA7M.mjs";
import "./chunk-DQSHJLUH.mjs";
import {
  JiraVulnerabilityAri
} from "./chunk-3SB2J6DP.mjs";
import "./chunk-2NDER42Y.mjs";
import {
  JiraSecurityWorkspaceAri
} from "./chunk-G6K6IBMD.mjs";
import "./chunk-7PTBO36C.mjs";
import {
  JiraSiteAri
} from "./chunk-JGY2JGQW.mjs";
import "./chunk-XIFA6EHM.mjs";
import {
  JiraSprintAri
} from "./chunk-PBGBP5UH.mjs";
import "./chunk-FGZEBJS7.mjs";
import {
  JiraUserBroadcastMessageAri
} from "./chunk-SWQH6UTV.mjs";
import "./chunk-MHNZKONH.mjs";
import {
  JiraResourceUsageRecommendationAri
} from "./chunk-YQRTNTMT.mjs";
import "./chunk-D7TMYPCS.mjs";
import {
  JiraRoleAri
} from "./chunk-Q3HFLFK2.mjs";
import "./chunk-WAQ2TE25.mjs";
import {
  JiraSecurityContainerAri
} from "./chunk-YW3TJHDE.mjs";
import "./chunk-4RYB4RHN.mjs";
import {
  JiraSecurityLevelAri
} from "./chunk-VZ3A3L66.mjs";
import "./chunk-PO2YLXFH.mjs";
import {
  JiraRemoteLinkAri
} from "./chunk-BLB5SHDU.mjs";
import "./chunk-KHCVG4TJ.mjs";
import {
  JiraRepositoryAri
} from "./chunk-DRO4OS5T.mjs";
import "./chunk-PYOC3DJR.mjs";
import {
  JiraResolutionAri
} from "./chunk-FP426GZW.mjs";
import "./chunk-BOQOC7ST.mjs";
import {
  JiraResourceUsageMetricAri
} from "./chunk-TS3O4SGC.mjs";
import "./chunk-LLLNN75K.mjs";
import {
  JiraProjectAri
} from "./chunk-VDFLZOOK.mjs";
import "./chunk-CQAE6WHE.mjs";
import {
  JiraPullRequestHistoryAri
} from "./chunk-B5DZUQ2B.mjs";
import "./chunk-IBNDOKW7.mjs";
import {
  JiraPullRequestAri
} from "./chunk-IGUNXKYW.mjs";
import "./chunk-5FKWU5BE.mjs";
import {
  JiraRemoteLinkHistoryAri
} from "./chunk-ZQJ7CBHR.mjs";
import "./chunk-QY36TOVX.mjs";
import {
  JiraProjectOverviewAri
} from "./chunk-SIM3XAG6.mjs";
import "./chunk-ORHZXN2K.mjs";
import {
  JiraProjectRoleActorAri
} from "./chunk-BXLUGRFD.mjs";
import "./chunk-6XVY5ZJP.mjs";
import {
  JiraProjectShortcutAri
} from "./chunk-TM6B6YM2.mjs";
import "./chunk-MAZNZE5K.mjs";
import {
  JiraProjectTypeAri
} from "./chunk-GXDXIESF.mjs";
import "./chunk-OF3K3RDZ.mjs";
import {
  JiraPostIncidentReviewLinkAri
} from "./chunk-6GUD6ORZ.mjs";
import "./chunk-WSTI2GHH.mjs";
import {
  JiraPriorityAri
} from "./chunk-OFMQK4BB.mjs";
import "./chunk-C4M7DMM2.mjs";
import {
  JiraProjectCategoryAri
} from "./chunk-GR7APMAD.mjs";
import "./chunk-23L4LUZM.mjs";
import {
  JiraProjectFeatureAri
} from "./chunk-WIHSD6UB.mjs";
import "./chunk-QKR33HHC.mjs";
import {
  JiraOperationsComponentAri
} from "./chunk-RS2IWKXH.mjs";
import "./chunk-PHMOFVGN.mjs";
import {
  JiraOperationsWorkspaceAri
} from "./chunk-4E5RQTOE.mjs";
import "./chunk-QTWVEF4M.mjs";
import {
  JiraPermissionSchemeAri
} from "./chunk-ONQABU7V.mjs";
import "./chunk-SASLBDN4.mjs";
import {
  JiraPlanAri
} from "./chunk-5CEYYAEJ.mjs";
import "./chunk-GGTGTA2K.mjs";
import {
  JiraIssueAri
} from "./chunk-O3NE3WVS.mjs";
import "./chunk-6MDCCCXR.mjs";
import {
  JiraIssuefieldvalueAri
} from "./chunk-SRXGCIKX.mjs";
import "./chunk-ZNA5VEMQ.mjs";
import {
  JiraNotificationTypeSchemeAri
} from "./chunk-YZDCNEC7.mjs";
import "./chunk-45SACCRI.mjs";
import {
  JiraNotificationTypeAri
} from "./chunk-UWZM2ARE.mjs";
import "./chunk-QXA66ZDW.mjs";
import {
  JiraIssueTypeSchemeMappingAri
} from "./chunk-T33INYRB.mjs";
import "./chunk-VNWAD3NX.mjs";
import {
  JiraIssueTypeSchemeAri
} from "./chunk-KUHP7EWE.mjs";
import "./chunk-QFTRTL74.mjs";
import {
  JiraIssueTypeAri
} from "./chunk-QXUGVB63.mjs";
import "./chunk-YBV4ZT7H.mjs";
import {
  JiraIssueWorklogAri
} from "./chunk-STOR2I4I.mjs";
import "./chunk-O6FNFDKY.mjs";
import {
  JiraIssueHistoryAri
} from "./chunk-JNMLTN7N.mjs";
import "./chunk-N64LAJN2.mjs";
import {
  JiraIssueLinkTypeAri
} from "./chunk-HIVMFFMD.mjs";
import "./chunk-ZJWWJNVZ.mjs";
import {
  JiraIssueSearchViewAri
} from "./chunk-BUE6YJLP.mjs";
import "./chunk-4BHYFYKX.mjs";
import {
  JiraIssueStatusAri
} from "./chunk-5KNIC4WB.mjs";
import "./chunk-S3WTNWJL.mjs";
import {
  JiraIssueAttachmentAri
} from "./chunk-WEQM2RED.mjs";
import "./chunk-AU74CHJB.mjs";
import {
  JiraIssueCommentAri
} from "./chunk-AI7AMMDE.mjs";
import "./chunk-C7LAHHR5.mjs";
import {
  JiraIssueFieldMetadataAri
} from "./chunk-7QGAZE5P.mjs";
import "./chunk-Q3PRSOZC.mjs";
import {
  JiraIssueFieldOptionAri
} from "./chunk-NQLRPD4M.mjs";
import "./chunk-TKZ6M5JE.mjs";
import {
  JiraFilterEmailSubscriptionAri
} from "./chunk-R25BVPWY.mjs";
import "./chunk-Z6KRUHKY.mjs";
import {
  JiraFilterAri
} from "./chunk-OYMZNIVT.mjs";
import "./chunk-NQBZEVCS.mjs";
import {
  JiraIncidentHistoryAri
} from "./chunk-6IZXRNNU.mjs";
import "./chunk-7AJWNJ45.mjs";
import {
  JiraIncidentAri
} from "./chunk-ROML2EZA.mjs";
import "./chunk-5F3IR2Z7.mjs";
import {
  JiraEntityPropertyAri
} from "./chunk-BJNPREJK.mjs";
import "./chunk-WVAVMODN.mjs";
import {
  JiraFavouriteAri
} from "./chunk-LTMNQDCN.mjs";
import "./chunk-ML5L5N3V.mjs";
import {
  JiraFeatureFlagHistoryAri
} from "./chunk-XIFTKDUD.mjs";
import "./chunk-BLEHWG53.mjs";
import {
  JiraFeatureFlagAri
} from "./chunk-4IGHHYJ2.mjs";
import "./chunk-KSD4G6QQ.mjs";
import {
  JiraDesignHistoryAri
} from "./chunk-ZHRCWPGQ.mjs";
import "./chunk-UG2AH2TA.mjs";
import {
  JiraDesignAri
} from "./chunk-KSA2UXMR.mjs";
import "./chunk-DSHHZHIG.mjs";
import {
  JiraDocumentHistoryAri
} from "./chunk-WNHA7WZH.mjs";
import "./chunk-Z2JLOUYO.mjs";
import {
  JiraDocumentAri
} from "./chunk-VWTP4Z4G.mjs";
import "./chunk-LUBY6O72.mjs";
import {
  JiraConnectAppAri
} from "./chunk-DTECF7E7.mjs";
import "./chunk-VRZRWJMV.mjs";
import {
  JiraDashboardAri
} from "./chunk-PE737NLA.mjs";
import "./chunk-KZLF4LHF.mjs";
import {
  JiraDeploymentHistoryAri
} from "./chunk-BYM3FIVJ.mjs";
import "./chunk-UD7ZCFKR.mjs";
import {
  JiraDeploymentAri
} from "./chunk-WLOYKF7J.mjs";
import "./chunk-ACU6GJRH.mjs";
import {
  JiraBulkOperationTaskAri
} from "./chunk-PSYKQWAG.mjs";
import "./chunk-Z76VAYJY.mjs";
import {
  JiraCommentAri
} from "./chunk-YEN3Y4BC.mjs";
import "./chunk-UWONVETB.mjs";
import {
  JiraCommitAri
} from "./chunk-YE5QICX2.mjs";
import "./chunk-YBNRWWV7.mjs";
import {
  JiraComponentAri
} from "./chunk-6DQOOMDQ.mjs";
import "./chunk-DUSRJMGF.mjs";
import {
  JiraBranchHistoryAri
} from "./chunk-TDGBIMH4.mjs";
import "./chunk-LZMQ2B5J.mjs";
import {
  JiraBranchAri
} from "./chunk-FTR5XLBL.mjs";
import "./chunk-65K6V5WX.mjs";
import {
  JiraBuildHistoryAri
} from "./chunk-TJ7ABDJ7.mjs";
import "./chunk-55Z3USGX.mjs";
import {
  JiraBuildAri
} from "./chunk-JT46W4GF.mjs";
import "./chunk-I56NWCQF.mjs";
import {
  JiraAnnouncementBannerAri
} from "./chunk-24ZBCNZY.mjs";
import "./chunk-VFZPJ7ZD.mjs";
import {
  JiraAvailableFieldsAri
} from "./chunk-ZBE45CFD.mjs";
import "./chunk-ICE3OR6G.mjs";
import "./chunk-PU63LMKI.mjs";
import {
  JiraSoftwareUserBoardConfigAri
} from "./chunk-5TJFVDN7.mjs";
import "./chunk-3UUDXEXS.mjs";
import "./chunk-SV7H6YFL.mjs";
import "./chunk-ZOFOPGOV.mjs";
import {
  JiraSoftwareCustomFilterAri
} from "./chunk-ZWY4WXHZ.mjs";
import "./chunk-O6OVDRBO.mjs";
import {
  JiraSoftwareRoleAri
} from "./chunk-54FWRK5X.mjs";
import "./chunk-SKSGTSPT.mjs";
import {
  JiraSoftwareSiteAri
} from "./chunk-S3CNVRP2.mjs";
import "./chunk-CKKWVELS.mjs";
import {
  JiraSoftwareBoardAri
} from "./chunk-RREXEGYM.mjs";
import "./chunk-YJAJS2GF.mjs";
import {
  JiraSoftwareCardParentAri
} from "./chunk-FYL43QS3.mjs";
import "./chunk-M7R3FF5D.mjs";
import {
  JiraSoftwareCardAri
} from "./chunk-FEVJVZSE.mjs";
import "./chunk-6WK3XU6H.mjs";
import {
  JiraSoftwareColumnAri
} from "./chunk-YL7GKXTY.mjs";
import "./chunk-P7LHCIIY.mjs";
import {
  JiraSoftwareBoardFeatureAri
} from "./chunk-QI24Y2CQ.mjs";
import "./chunk-E24RP2LJ.mjs";
import {
  JiraSoftwareBoardIssueListAri
} from "./chunk-LD75KDIQ.mjs";
import "./chunk-3Z6ZDJ2A.mjs";
import "./chunk-WWZKXHXC.mjs";
import {
  JiraServicedeskSiteAri
} from "./chunk-DJWGAIEC.mjs";
import "./chunk-F4LW6XYO.mjs";
import {
  JiraServicedeskSlaAri
} from "./chunk-DR7P477K.mjs";
import "./chunk-BMLW67PZ.mjs";
import {
  JiraServicedeskOrganizationAri
} from "./chunk-ALWHCTLB.mjs";
import "./chunk-FYLFP7EF.mjs";
import {
  JiraServicedeskQueueAri
} from "./chunk-62PFEYHR.mjs";
import "./chunk-TC6T4VFX.mjs";
import {
  JiraServicedeskRequestTypeAri
} from "./chunk-N664AMAE.mjs";
import "./chunk-5SWOR5KD.mjs";
import {
  JiraServicedeskRoleAri
} from "./chunk-Z7CAEGFD.mjs";
import "./chunk-VV22WWWI.mjs";
import {
  JiraServicedeskApprovalAri
} from "./chunk-3T56YLWK.mjs";
import "./chunk-AZJHCKNE.mjs";
import {
  JiraServicedeskCalendarEventAri
} from "./chunk-LM3PMHC6.mjs";
import "./chunk-SMZUBZEM.mjs";
import {
  JiraServicedeskCannedResponseAri
} from "./chunk-W6BHJIAO.mjs";
import "./chunk-XGGNYA2D.mjs";
import "./chunk-ENGY2TDF.mjs";
import {
  JiraAlignInstanceAri
} from "./chunk-ZTXBYNEM.mjs";
import "./chunk-KIGWQUKL.mjs";
import "./chunk-QAQMSQTV.mjs";
import {
  JiraCoreSiteAri
} from "./chunk-AFF5IQCJ.mjs";
import "./chunk-BYLK2PHJ.mjs";
import {
  JiraCoreRoleAri
} from "./chunk-UQYRDHTA.mjs";
import "./chunk-22MNSNE6.mjs";
import "./chunk-7ZZEN3W6.mjs";
import {
  IdentityTeamAri
} from "./chunk-3RPSDZ43.mjs";
import "./chunk-T76PG5FI.mjs";
import {
  IdentityUserAri
} from "./chunk-2QYQOO4P.mjs";
import "./chunk-N5QB45QP.mjs";
import {
  IdentityUserGrantAri
} from "./chunk-4U7BW5PZ.mjs";
import "./chunk-7Y76725A.mjs";
import {
  IdentityUserbaseAri
} from "./chunk-OD5QU56O.mjs";
import "./chunk-FKBB466V.mjs";
import {
  IdentityOauthClientAri
} from "./chunk-BA3MXQGL.mjs";
import "./chunk-UTUDGZLV.mjs";
import {
  IdentityRoleAri
} from "./chunk-OBMDVOPU.mjs";
import "./chunk-UFIIDAET.mjs";
import {
  IdentitySiteAri
} from "./chunk-VG4AQW3G.mjs";
import "./chunk-CXXH2SWL.mjs";
import {
  IdentityTeamMemberAri
} from "./chunk-O424T4F3.mjs";
import "./chunk-OULHQOIR.mjs";
import {
  IdentityCustomerOrganizationAri
} from "./chunk-FC5AF6SZ.mjs";
import "./chunk-XIBIRHB4.mjs";
import {
  IdentityDirectoryAri
} from "./chunk-5YQVOGZF.mjs";
import "./chunk-L36L7XSX.mjs";
import {
  IdentityDomainAri
} from "./chunk-CT2Q3GUC.mjs";
import "./chunk-S5UAML7R.mjs";
import {
  IdentityGroupAri
} from "./chunk-IR5X2TXS.mjs";
import "./chunk-IZPJNHZU.mjs";
import {
  IdentityAppAri
} from "./chunk-UKOMUDB7.mjs";
import "./chunk-QULKIQO5.mjs";
import {
  IdentityAuthPolicyAri
} from "./chunk-WONGJX7R.mjs";
import "./chunk-27IAW7JZ.mjs";
import {
  IdentityCustomerDirectoryAri
} from "./chunk-NXHTTLRT.mjs";
import "./chunk-KQGM5YYP.mjs";
import "./chunk-BQGCFS74.mjs";
import {
  HelpRequestFormAri
} from "./chunk-SDD63CEL.mjs";
import "./chunk-QF6HVRWR.mjs";
import {
  HelpHelpDeskAri
} from "./chunk-OFMDHRK5.mjs";
import "./chunk-EQROJBZ3.mjs";
import {
  HelpLayoutAri
} from "./chunk-4USWVLEC.mjs";
import "./chunk-HXZGYRZT.mjs";
import {
  HelpPortalAri
} from "./chunk-Z53VRBCD.mjs";
import "./chunk-DEFOWKTU.mjs";
import {
  HelpArticleAri
} from "./chunk-SWRRQYMY.mjs";
import "./chunk-RDV7AM67.mjs";
import {
  HelpChannelAri
} from "./chunk-YJK2L47S.mjs";
import "./chunk-H2XK65KU.mjs";
import {
  HelpHelpCenterAri
} from "./chunk-72GQLWK4.mjs";
import "./chunk-UIZK5L5W.mjs";
import "./chunk-KNBR3WSA.mjs";
import {
  GraphWorkspaceAri
} from "./chunk-3IPCLHZX.mjs";
import "./chunk-SQM2NR5M.mjs";
import {
  GraphServiceRelationshipAri
} from "./chunk-V3IYIS6Q.mjs";
import "./chunk-JNGK7AYM.mjs";
import {
  GraphServiceAri
} from "./chunk-W2BL2AH6.mjs";
import "./chunk-KKDL7RK7.mjs";
import {
  GraphVulnerabilityHistoryAri
} from "./chunk-G57SGWL4.mjs";
import "./chunk-PHKBIUQV.mjs";
import {
  GraphVulnerabilityAri
} from "./chunk-DF53FWVZ.mjs";
import "./chunk-J7VD66AE.mjs";
import {
  GraphSecurityWorkspaceAri
} from "./chunk-AC742ZQS.mjs";
import "./chunk-4FPKBG6M.mjs";
import {
  GraphServiceAndJiraProjectRelationshipAri
} from "./chunk-Y3RTRVZV.mjs";
import "./chunk-723KYNJS.mjs";
import {
  GraphServiceAndOperationsTeamRelationshipAri
} from "./chunk-OXL22VVL.mjs";
import "./chunk-E4XM6JBY.mjs";
import {
  GraphServiceAndVcsRepositoryRelationshipAri
} from "./chunk-L6HAERSK.mjs";
import "./chunk-SYSNNBBE.mjs";
import {
  GraphRemoteLinkHistoryAri
} from "./chunk-IODN6B4N.mjs";
import "./chunk-2YKXM3UN.mjs";
import {
  GraphRemoteLinkAri
} from "./chunk-SGWD7EQZ.mjs";
import "./chunk-7CH7D4HJ.mjs";
import {
  GraphRepositoryAri
} from "./chunk-N6E4JHM6.mjs";
import "./chunk-SAJO4IGE.mjs";
import {
  GraphSecurityContainerAri
} from "./chunk-ZVUQSFXL.mjs";
import "./chunk-6HYQV6SW.mjs";
import {
  GraphJiraProjectAndVcsRepositoryRelationshipAri
} from "./chunk-QF6WLUXE.mjs";
import "./chunk-XYDSJQO4.mjs";
import {
  GraphOperationsWorkspaceAri
} from "./chunk-H7NJT7NT.mjs";
import "./chunk-5HHNCVXF.mjs";
import {
  GraphPullRequestHistoryAri
} from "./chunk-VGQ2PY3T.mjs";
import "./chunk-K4SS2RGK.mjs";
import {
  GraphPullRequestAri
} from "./chunk-LS7IS6UB.mjs";
import "./chunk-Z3FWECDS.mjs";
import {
  GraphJiraProjectAndDevopsToolRelationshipAri
} from "./chunk-5Z2QKNXU.mjs";
import "./chunk-2A255S76.mjs";
import {
  GraphJiraProjectAndDocumentationSpaceRelationshipAri
} from "./chunk-ICORSST3.mjs";
import "./chunk-SM4OYQAG.mjs";
import {
  GraphJiraProjectAndOperationsTeamRelationshipAri
} from "./chunk-4DDRLSRP.mjs";
import "./chunk-35GRSR7B.mjs";
import "./chunk-JYQ5JCSW.mjs";
import "./chunk-IUR7MTIZ.mjs";
import {
  GraphFeatureFlagHistoryAri
} from "./chunk-6EZCVVRT.mjs";
import "./chunk-DFRD6YBM.mjs";
import {
  GraphFeatureFlagAri
} from "./chunk-4T23K4P2.mjs";
import "./chunk-LDUD7O2G.mjs";
import {
  GraphIncidentHistoryAri
} from "./chunk-ZYBRRGV4.mjs";
import "./chunk-NTBEFMSE.mjs";
import {
  GraphIncidentAri
} from "./chunk-5G37SQYR.mjs";
import "./chunk-FKF6Z5DH.mjs";
import {
  GraphDesignHistoryAri
} from "./chunk-UST7DEEU.mjs";
import "./chunk-MTSTC2DF.mjs";
import {
  GraphDesignAri
} from "./chunk-RPVFQH7A.mjs";
import "./chunk-RBS5RYKM.mjs";
import {
  GraphDocumentHistoryAri
} from "./chunk-T7RJ6LYW.mjs";
import "./chunk-SP4WD2VN.mjs";
import {
  GraphDocumentAri
} from "./chunk-AHUI46I6.mjs";
import "./chunk-J4TEUSCJ.mjs";
import {
  GraphBuildAri
} from "./chunk-BAKXQOM7.mjs";
import "./chunk-5YTWJ4WK.mjs";
import {
  GraphCommitAri
} from "./chunk-GB7Q3NQ5.mjs";
import "./chunk-LJ7XURCE.mjs";
import {
  GraphDeploymentHistoryAri
} from "./chunk-AJZLGKPR.mjs";
import "./chunk-LCOEZ74N.mjs";
import {
  GraphDeploymentAri
} from "./chunk-ENLF7OFK.mjs";
import "./chunk-OON2ZRPC.mjs";
import {
  GraphBranchHistoryAri
} from "./chunk-6OIVMFVC.mjs";
import "./chunk-RL6RLD25.mjs";
import {
  GraphBranchAri
} from "./chunk-QLTAVMFU.mjs";
import "./chunk-JJHDDJHM.mjs";
import {
  GraphBuildHistoryAri
} from "./chunk-M6TKNOHJ.mjs";
import "./chunk-Q4A3TDFH.mjs";
import "./chunk-5PRHJDN3.mjs";
import "./chunk-3DLTQ4XM.mjs";
import {
  GoogleSpreadsheetAri
} from "./chunk-OH62MPVY.mjs";
import "./chunk-466Z3YAU.mjs";
import {
  GoogleDocumentAri
} from "./chunk-V7WAPEBC.mjs";
import "./chunk-5VDZQSQU.mjs";
import {
  GoogleFormAri
} from "./chunk-IUXNBASD.mjs";
import "./chunk-X63IP3OI.mjs";
import {
  GooglePresentationAri
} from "./chunk-BKJJDB6P.mjs";
import "./chunk-5B6S6SWP.mjs";
import "./chunk-ZMVXG3Z6.mjs";
import {
  ExusExternalUserAri
} from "./chunk-LSGWQXD2.mjs";
import "./chunk-IWWNXWBX.mjs";
import {
  FigmaFileAri
} from "./chunk-EP5JAQW2.mjs";
import "./chunk-4R4BPJCR.mjs";
import "./chunk-YHZUH32W.mjs";
import {
  EcosystemExtensionGroupAri
} from "./chunk-NONSJJMI.mjs";
import "./chunk-NYRDQJIQ.mjs";
import {
  EcosystemExtensionAri
} from "./chunk-RYWU5YFJ.mjs";
import "./chunk-Q6OH5L4G.mjs";
import "./chunk-J6TNPU7E.mjs";
import {
  ElementsWorkspaceAri
} from "./chunk-ZMG5FBV2.mjs";
import "./chunk-5TZRCHUN.mjs";
import {
  ElementsCustomEmojiMetadataAri
} from "./chunk-TAJVOUEK.mjs";
import "./chunk-INZ6GYXP.mjs";
import {
  EcosystemAppEnvironmentVersionAri
} from "./chunk-JRTYCYT2.mjs";
import "./chunk-CD56DLNS.mjs";
import {
  EcosystemAppAri
} from "./chunk-PNGCHIF2.mjs";
import "./chunk-ZSCWAZK4.mjs";
import {
  EcosystemConnectAppAri
} from "./chunk-JBMJVRQK.mjs";
import "./chunk-GEYJT4UH.mjs";
import {
  EcosystemEnvironmentAri
} from "./chunk-ZH2KTYPV.mjs";
import "./chunk-UJZYZKMO.mjs";
import "./chunk-PQ7YGB4H.mjs";
import "./chunk-Q676EFQB.mjs";
import {
  DataLakeLakeConnectionAri
} from "./chunk-MNEWE6O7.mjs";
import "./chunk-2Y4LARQO.mjs";
import "./chunk-PTM25UW5.mjs";
import {
  DevopsNamespaceAri
} from "./chunk-Y5DNILHP.mjs";
import "./chunk-H53KN2GD.mjs";
import {
  DevopsProviderAri
} from "./chunk-LOUKCKZH.mjs";
import "./chunk-OILGX3NZ.mjs";
import {
  DevopsRelationshipAri
} from "./chunk-M4QVWM3U.mjs";
import "./chunk-PPXOGRS2.mjs";
import {
  DevopsToolAri
} from "./chunk-V456L5PW.mjs";
import "./chunk-WECRO6DG.mjs";
import {
  DevopsContainerAri
} from "./chunk-HYYYGQGJ.mjs";
import "./chunk-FGEFELP3.mjs";
import {
  ConversationalHelpConversationMessageAri
} from "./chunk-HF76CX6B.mjs";
import "./chunk-W6PEKGFC.mjs";
import {
  ConversationalHelpConversationAri
} from "./chunk-KXCSGCY2.mjs";
import "./chunk-YWVV6JNH.mjs";
import "./chunk-VZG2Z6DB.mjs";
import {
  ConfluenceWhiteboardAri
} from "./chunk-5BKGXKQW.mjs";
import "./chunk-SRDRWGU4.mjs";
import {
  ConfluenceWorkspaceAri
} from "./chunk-MEDW5AYO.mjs";
import "./chunk-2BXLBFIW.mjs";
import {
  ConfluenceSpaceAri
} from "./chunk-SV5ZOKQT.mjs";
import "./chunk-UOPZEJBG.mjs";
import {
  ConfluenceTaskAri
} from "./chunk-HKBVMC2I.mjs";
import "./chunk-SHZM4TTX.mjs";
import {
  ConfluenceTeamCalendarAri
} from "./chunk-YSQD3424.mjs";
import "./chunk-KKW3GIW7.mjs";
import {
  ConfluenceUserPropertyAri
} from "./chunk-OXDVGNLG.mjs";
import "./chunk-NW7N5UEI.mjs";
import {
  ConfluenceNoteAri
} from "./chunk-PEUFHEUT.mjs";
import "./chunk-ZWIRILL2.mjs";
import {
  ConfluencePageAri
} from "./chunk-WCGOKX23.mjs";
import "./chunk-IMF5GHHQ.mjs";
import {
  ConfluenceRoleAri
} from "./chunk-AA2FJEJD.mjs";
import "./chunk-VTWZOXAB.mjs";
import {
  ConfluenceSiteAri
} from "./chunk-XS663SLQ.mjs";
import "./chunk-PVOLGFB2.mjs";
import {
  ConfluenceDatabaseAri
} from "./chunk-OBJVV47G.mjs";
import "./chunk-PGMODI5S.mjs";
import "./chunk-K2XTPA47.mjs";
import "./chunk-N5UKNFK5.mjs";
import {
  ConfluenceEventAri
} from "./chunk-LOF3QY52.mjs";
import "./chunk-GWVYXALY.mjs";
import {
  ConfluenceLongRunningTaskAri
} from "./chunk-IPDWD3R5.mjs";
import "./chunk-E6K4JCID.mjs";
import {
  ConfluenceAttachmentAri
} from "./chunk-YB7ZPDHR.mjs";
import "./chunk-QCPR2D6J.mjs";
import {
  ConfluenceBlogpostAri
} from "./chunk-ZM54V656.mjs";
import "./chunk-J5SGYOAJ.mjs";
import {
  ConfluenceCommentAri
} from "./chunk-SXAP7XJO.mjs";
import "./chunk-ARFRFJQD.mjs";
import {
  ConfluenceContentAri
} from "./chunk-2RGHGCFP.mjs";
import "./chunk-LVQBHR63.mjs";
import {
  ConfluenceAdminAnnouncementBannerAri
} from "./chunk-Y3RPUOW4.mjs";
import "./chunk-IH6QYLXI.mjs";
import "./chunk-BEV3P7ZD.mjs";
import {
  CompassUserDefinedParameterAri
} from "./chunk-DMFJWHOA.mjs";
import "./chunk-U7N5T7JC.mjs";
import {
  CompassWebhookAri
} from "./chunk-ZI76YP5W.mjs";
import "./chunk-RX4T5I6M.mjs";
import {
  CompassWorkspaceAri
} from "./chunk-N6UTJOZC.mjs";
import "./chunk-QQTPW4TZ.mjs";
import {
  CompassMetricSourceAri
} from "./chunk-U4ZH6ZBB.mjs";
import "./chunk-H63VXJES.mjs";
import {
  CompassRoleAri
} from "./chunk-XRQG6TJK.mjs";
import "./chunk-FJ7KFVKZ.mjs";
import {
  CompassScorecardAri
} from "./chunk-PCU7FWYO.mjs";
import "./chunk-2TSZWIWR.mjs";
import {
  CompassSiteAri
} from "./chunk-EDPSDGXF.mjs";
import "./chunk-SOJW4UXZ.mjs";
import {
  CompassEventSourceAri
} from "./chunk-RFKGTCZC.mjs";
import "./chunk-G6Y66A5E.mjs";
import {
  CompassIncomingWebhookAri
} from "./chunk-OXP565G7.mjs";
import "./chunk-6S4G5T3V.mjs";
import {
  CompassMetricDefinitionAri
} from "./chunk-PSJ56PXK.mjs";
import "./chunk-BS6YPB3T.mjs";
import {
  CompassComponentLabelAri
} from "./chunk-ZXSOSOY5.mjs";
import "./chunk-QOMYCLGO.mjs";
import {
  CompassComponentLinkAri
} from "./chunk-7MXGNWNL.mjs";
import "./chunk-RUUZK374.mjs";
import {
  CompassComponentAri
} from "./chunk-25S7IOCA.mjs";
import "./chunk-XW2UZKUO.mjs";
import {
  CompassCustomFieldDefinitionAri
} from "./chunk-C5TDU5JL.mjs";
import "./chunk-K3BVQB6G.mjs";
import {
  CompassAssistantAnswerAri
} from "./chunk-6IWA2LKS.mjs";
import "./chunk-VAWIKJGT.mjs";
import "./chunk-NWYLOPBR.mjs";
import {
  CommerceShipToPartyAri
} from "./chunk-2TO2SAAN.mjs";
import "./chunk-MR7K6E3K.mjs";
import {
  CommerceTransactionAccountAri
} from "./chunk-6HAR4ROV.mjs";
import "./chunk-IG6GXX55.mjs";
import {
  CommerceWorkbenchResourceAri
} from "./chunk-ADMBEYCT.mjs";
import "./chunk-CVI37ASZ.mjs";
import {
  CommerceProductAri
} from "./chunk-TW7SGTSZ.mjs";
import "./chunk-7KRTMBHT.mjs";
import {
  CommercePromotionAri
} from "./chunk-6QO3EJAX.mjs";
import "./chunk-NDIFFXAM.mjs";
import {
  CommerceQuoteAri
} from "./chunk-CEGBRX5N.mjs";
import "./chunk-2PBMRYJM.mjs";
import {
  CommerceRoleAri
} from "./chunk-OOEEWONU.mjs";
import "./chunk-T3FMJP7C.mjs";
import {
  CommerceOfferingAri
} from "./chunk-E2X6SN34.mjs";
import "./chunk-ANF2XXE7.mjs";
import {
  CommerceOrderAri
} from "./chunk-DK4M7FFC.mjs";
import "./chunk-3AXUCZXV.mjs";
import {
  CommercePaymentMethodAri
} from "./chunk-5NWXOTQQ.mjs";
import "./chunk-N3JGQE6H.mjs";
import {
  CommercePricingPlanAri
} from "./chunk-JJJEWQFH.mjs";
import "./chunk-66TMLCVM.mjs";
import {
  CommerceEntitlementAri
} from "./chunk-QH34FSGN.mjs";
import "./chunk-NINDB5F3.mjs";
import {
  CommerceInitiativeAri
} from "./chunk-ENVJ6QH5.mjs";
import "./chunk-NEKFBJUG.mjs";
import {
  CommerceInvoiceGroupAri
} from "./chunk-GKHZ4KWD.mjs";
import "./chunk-MX6GMSIL.mjs";
import {
  CommerceOfferingRelationshipTemplateAri
} from "./chunk-PQSCYVGF.mjs";
import "./chunk-2P7D7BWD.mjs";
import {
  CommerceCatalogAccountAri
} from "./chunk-KIP76WC7.mjs";
import "./chunk-3FQB6Q73.mjs";
import {
  CommerceChangeAri
} from "./chunk-HG4CLAAH.mjs";
import "./chunk-R3MD4VCL.mjs";
import {
  CommerceEntitlementTemplateAri
} from "./chunk-XAZDDRTI.mjs";
import "./chunk-32JDQQPU.mjs";
import "./chunk-7USM4R6S.mjs";
import {
  CmdbWorkspaceAri
} from "./chunk-U4FVSEKE.mjs";
import "./chunk-UQ5PWXBJ.mjs";
import {
  CmdbObjectAri
} from "./chunk-ENA2HPUD.mjs";
import "./chunk-BZ6D72WR.mjs";
import "./chunk-6NZP4J6O.mjs";
import "./chunk-WF26OXZU.mjs";
import {
  CmdbSchemaAri
} from "./chunk-5ONGUVVK.mjs";
import "./chunk-7NF55AHC.mjs";
import {
  CmdbTypeAri
} from "./chunk-RAV3YQAK.mjs";
import "./chunk-Q5AHPTNW.mjs";
import {
  CmdbAttributeAri
} from "./chunk-UPBYPKVR.mjs";
import "./chunk-XAMITGNQ.mjs";
import {
  CmdbImportConfigurationAri
} from "./chunk-G7EDH3DR.mjs";
import "./chunk-W3AHDHPL.mjs";
import "./chunk-KLYBIY73.mjs";
import {
  CanvasWorkspaceAri
} from "./chunk-IFIZHV5Q.mjs";
import "./chunk-RSJD3TU6.mjs";
import {
  CanvasBoardAri
} from "./chunk-W45NQ4BU.mjs";
import "./chunk-R3VA6JMD.mjs";
import {
  CanvasDatabaseAri
} from "./chunk-U6BUCSWR.mjs";
import "./chunk-LRE2PRU7.mjs";
import {
  CanvasSiteAri
} from "./chunk-PNXPIH25.mjs";
import "./chunk-VCVOT2RL.mjs";
import "./chunk-QKWQ73L2.mjs";
import {
  BrieSiteAri
} from "./chunk-ZMKHWWHO.mjs";
import "./chunk-2CDIPTTF.mjs";
import {
  BrieRoleAri
} from "./chunk-J4N7XZSI.mjs";
import "./chunk-S5IRDM4L.mjs";
import "./chunk-CI2TIA3X.mjs";
import {
  BitbucketTeamAri
} from "./chunk-25JXJPOM.mjs";
import "./chunk-O3ROTQWU.mjs";
import {
  BitbucketWorkspaceAri
} from "./chunk-E5SVU57T.mjs";
import "./chunk-A35PGQYK.mjs";
import {
  BitbucketPullrequestAri
} from "./chunk-BNC46TWT.mjs";
import "./chunk-7V62G36F.mjs";
import {
  BitbucketRepositoryTransferAri
} from "./chunk-AL5552Z4.mjs";
import "./chunk-QL4XDR5W.mjs";
import {
  BitbucketRepositoryAri
} from "./chunk-S44QJPFM.mjs";
import "./chunk-4KEXHL4E.mjs";
import {
  BitbucketSiteAri
} from "./chunk-IUJXK5LD.mjs";
import "./chunk-NKJWWYMZ.mjs";
import {
  BitbucketAppAri
} from "./chunk-FNAUAADG.mjs";
import "./chunk-HN5XXUJU.mjs";
import {
  BitbucketOauthConsumerAri
} from "./chunk-V4YT66GO.mjs";
import "./chunk-B6DXOH6V.mjs";
import {
  BitbucketProjectAri
} from "./chunk-A3YCULML.mjs";
import "./chunk-KZCUOZCO.mjs";
import "./chunk-RLLBEQ3Y.mjs";
import {
  BeaconWorkspaceAri
} from "./chunk-36DUFOWZ.mjs";
import "./chunk-POXHAMNN.mjs";
import {
  BeaconRoleAri
} from "./chunk-RYOIMYPF.mjs";
import "./chunk-YRF7LNG2.mjs";
import {
  BeaconSiteAri
} from "./chunk-MIKOXDTA.mjs";
import "./chunk-JQQF6HSU.mjs";
import {
  BeaconSubscriptionAri
} from "./chunk-CPR767GV.mjs";
import "./chunk-CWWCGBU4.mjs";
import "./chunk-2L6JHSPV.mjs";
import {
  AvpRoleAri
} from "./chunk-HLNI2ECZ.mjs";
import "./chunk-SZNOZQ4T.mjs";
import {
  AvpSiteAri
} from "./chunk-FGOPWHXQ.mjs";
import "./chunk-GFZFXLJN.mjs";
import {
  BeaconAlertAri
} from "./chunk-LGL7MCFG.mjs";
import "./chunk-TJOHF5RH.mjs";
import {
  BeaconDetectionAri
} from "./chunk-UOBD5PNV.mjs";
import "./chunk-L27OOBSM.mjs";
import {
  AvpDashboardAri
} from "./chunk-E7N22YFC.mjs";
import "./chunk-L3TDTHN7.mjs";
import {
  AvpDatasourceAri
} from "./chunk-SERXRY6V.mjs";
import "./chunk-MIBCZ475.mjs";
import "./chunk-WW2ANN25.mjs";
import {
  AvocadoSiteAri
} from "./chunk-ZVQNSLOC.mjs";
import "./chunk-3G3BCF2A.mjs";
import {
  AvocadoUserAri
} from "./chunk-XSCNB3DN.mjs";
import "./chunk-3W3PTAEC.mjs";
import {
  AvocadoAnswerAri
} from "./chunk-ZUGZRJQE.mjs";
import "./chunk-MBGO4Q2M.mjs";
import {
  AvocadoQuestionAri
} from "./chunk-ICFSCYFU.mjs";
import "./chunk-URX7NO2Y.mjs";
import {
  AvocadoRoleAri
} from "./chunk-CN5AJH4V.mjs";
import "./chunk-DKX2RZ2Z.mjs";
import "./chunk-567EEHI7.mjs";
import {
  AtlashubSiteAri
} from "./chunk-JNS77PIY.mjs";
import "./chunk-SJUWQEXL.mjs";
import "./chunk-Z7UEZQHZ.mjs";
import {
  AutomationRoleAri
} from "./chunk-SUNICAQN.mjs";
import "./chunk-QUWGD5ND.mjs";
import {
  AutomationRuleAri
} from "./chunk-MKHEX6LK.mjs";
import "./chunk-U7AZ64SA.mjs";
import {
  Arm
} from "./chunk-LFU4YYFS.mjs";
import {
  AtlashubIdAri
} from "./chunk-IHU6DNJP.mjs";
import "./chunk-C4MJWUCX.mjs";
import "./chunk-P5EQPJFA.mjs";
import {
  AnalyticsUserAri
} from "./chunk-YZ2IC7TU.mjs";
import "./chunk-77DTJ44V.mjs";
import "./chunk-ZGGM6QS4.mjs";
import {
  ApprovalDeskWorkflowAri
} from "./chunk-ZTCJXDCQ.mjs";
import "./chunk-LIEBAJFZ.mjs";
import {
  ApprovalDeskWorkspaceAri
} from "./chunk-LGYQKVSX.mjs";
import "./chunk-YTLJFQ5C.mjs";
import {
  ApprovalDeskDecisionTableAri
} from "./chunk-ZJAFDA7S.mjs";
import "./chunk-GTNXGX3Z.mjs";
import {
  ApprovalDeskRoleAri
} from "./chunk-MB34E6AT.mjs";
import "./chunk-OQIMDPGX.mjs";
import "./chunk-PA4Y7TWI.mjs";
import {
  AnyAri,
  Ari
} from "./chunk-NCC2UXNN.mjs";
import {
  Ati
} from "./chunk-2UUYVBKT.mjs";
import "./chunk-GDQPXUOZ.mjs";
import {
  ValidationError
} from "./chunk-FAZYALZ2.mjs";

// src/core/types/deprecated/resource-owner-enum.ts
var ResourceOwnerEnum = /* @__PURE__ */ ((ResourceOwnerEnum2) => (ResourceOwnerEnum2.Analytics = "analytics", ResourceOwnerEnum2.ApprovalDesk = "approval-desk", ResourceOwnerEnum2.Atlashub = "atlashub", ResourceOwnerEnum2.Automation = "automation", ResourceOwnerEnum2.Avocado = "avocado", ResourceOwnerEnum2.Avp = "avp", ResourceOwnerEnum2.Beacon = "beacon", ResourceOwnerEnum2.Bitbucket = "bitbucket", ResourceOwnerEnum2.Brie = "brie", ResourceOwnerEnum2.Canvas = "canvas", ResourceOwnerEnum2.Cmdb = "cmdb", ResourceOwnerEnum2.Commerce = "commerce", ResourceOwnerEnum2.Compass = "compass", ResourceOwnerEnum2.Confluence = "confluence", ResourceOwnerEnum2.ConversationalHelp = "conversational-help", ResourceOwnerEnum2.DataLake = "data-lake", ResourceOwnerEnum2.Devops = "devops", ResourceOwnerEnum2.Ecosystem = "ecosystem", ResourceOwnerEnum2.Elements = "elements", ResourceOwnerEnum2.ExusExternal = "exus-external", ResourceOwnerEnum2.Figma = "figma", ResourceOwnerEnum2.Google = "google", ResourceOwnerEnum2.Graph = "graph", ResourceOwnerEnum2.Help = "help", ResourceOwnerEnum2.Identity = "identity", ResourceOwnerEnum2.Jira = "jira", ResourceOwnerEnum2.JiraAlign = "jira-align", ResourceOwnerEnum2.JiraCore = "jira-core", ResourceOwnerEnum2.JiraServicedesk = "jira-servicedesk", ResourceOwnerEnum2.JiraSoftware = "jira-software", ResourceOwnerEnum2.LinkingPlatform = "linking-platform", ResourceOwnerEnum2.MakerSpace = "maker-space", ResourceOwnerEnum2.Marketing = "marketing", ResourceOwnerEnum2.Measurement = "measurement", ResourceOwnerEnum2.Media = "media", ResourceOwnerEnum2.Mercury = "mercury", ResourceOwnerEnum2.Oauth = "oauth", ResourceOwnerEnum2.Opsgenie = "opsgenie", ResourceOwnerEnum2.Papi = "papi", ResourceOwnerEnum2.Passionfruit = "passionfruit", ResourceOwnerEnum2.PeoplePerftool = "people-perftool", ResourceOwnerEnum2.Platform = "platform", ResourceOwnerEnum2.PlatformServices = "platform-services", ResourceOwnerEnum2.RuntimeAuthClient = "runtime-auth-client", ResourceOwnerEnum2.Search = "search", ResourceOwnerEnum2.Statuspage = "statuspage", ResourceOwnerEnum2.Support = "support", ResourceOwnerEnum2.Teams = "teams", ResourceOwnerEnum2.Townsquare = "townsquare", ResourceOwnerEnum2.Trello = "trello", ResourceOwnerEnum2.UnifiedHelp = "unified-help", ResourceOwnerEnum2.VirtualAgent = "virtual-agent", ResourceOwnerEnum2))(ResourceOwnerEnum || {});

// src/core/types/deprecated/resource-type-enum.ts
var ResourceTypeEnum = /* @__PURE__ */ ((ResourceTypeEnum2) => (ResourceTypeEnum2.AccountLogin = "account-login", ResourceTypeEnum2.AccountSettings = "account-settings", ResourceTypeEnum2.AdminAnnouncementBanner = "admin-announcement-banner", ResourceTypeEnum2.Alert = "alert", ResourceTypeEnum2.AlertRecipientLink = "alert-recipient-link", ResourceTypeEnum2.AnnouncementBanner = "announcement-banner", ResourceTypeEnum2.Answer = "answer", ResourceTypeEnum2.Api = "api", ResourceTypeEnum2.ApiRequestMetric = "api-request-metric", ResourceTypeEnum2.App = "app", ResourceTypeEnum2.AppEnvironmentVersion = "app-environment-version", ResourceTypeEnum2.Approval = "approval", ResourceTypeEnum2.Article = "article", ResourceTypeEnum2.AssistantAnswer = "assistant-answer", ResourceTypeEnum2.Attachment = "attachment", ResourceTypeEnum2.Attribute = "attribute", ResourceTypeEnum2.AuthPolicy = "auth-policy", ResourceTypeEnum2.AvailableFields = "available-fields", ResourceTypeEnum2.Blogpost = "blogpost", ResourceTypeEnum2.Board = "board", ResourceTypeEnum2.BoardFeature = "board-feature", ResourceTypeEnum2.BoardIssueList = "board-issue-list", ResourceTypeEnum2.Branch = "branch", ResourceTypeEnum2.BranchHistory = "branch-history", ResourceTypeEnum2.Build = "build", ResourceTypeEnum2.BuildHistory = "build-history", ResourceTypeEnum2.BulkOperationTask = "bulk-operation-task", ResourceTypeEnum2.CalendarEvent = "calendar-event", ResourceTypeEnum2.Calibration = "calibration", ResourceTypeEnum2.CallRouting = "call-routing", ResourceTypeEnum2.CannedResponse = "canned-response", ResourceTypeEnum2.Card = "card", ResourceTypeEnum2.CardParent = "card-parent", ResourceTypeEnum2.CatalogAccount = "catalog-account", ResourceTypeEnum2.Change = "change", ResourceTypeEnum2.Channel = "channel", ResourceTypeEnum2.ClassificationTag = "classification-tag", ResourceTypeEnum2.Client = "client", ResourceTypeEnum2.Column = "column", ResourceTypeEnum2.Comment = "comment", ResourceTypeEnum2.Commit = "commit", ResourceTypeEnum2.Component = "component", ResourceTypeEnum2.ComponentLabel = "component-label", ResourceTypeEnum2.ComponentLink = "component-link", ResourceTypeEnum2.Configuration = "configuration", ResourceTypeEnum2.ConnectApp = "connect-app", ResourceTypeEnum2.Container = "container", ResourceTypeEnum2.Content = "content", ResourceTypeEnum2.Conversation = "conversation", ResourceTypeEnum2.ConversationMessage = "conversation-message", ResourceTypeEnum2.CustomEmojiMetadata = "custom-emoji-metadata", ResourceTypeEnum2.Customer = "customer", ResourceTypeEnum2.CustomerDirectory = "customer-directory", ResourceTypeEnum2.CustomerDomain = "customer-domain", ResourceTypeEnum2.CustomerOrganization = "customer-organization", ResourceTypeEnum2.CustomFieldDefinition = "custom-field-definition", ResourceTypeEnum2.CustomFilter = "custom-filter", ResourceTypeEnum2.CustomRole = "custom-role", ResourceTypeEnum2.Dashboard = "dashboard", ResourceTypeEnum2.Database = "database", ResourceTypeEnum2.Datasource = "datasource", ResourceTypeEnum2.DecisionTable = "decision-table", ResourceTypeEnum2.Deployment = "deployment", ResourceTypeEnum2.DeploymentHistory = "deployment-history", ResourceTypeEnum2.DeploymentServiceLink = "deployment-service-link", ResourceTypeEnum2.Design = "design", ResourceTypeEnum2.DesignHistory = "design-history", ResourceTypeEnum2.Detection = "detection", ResourceTypeEnum2.Directory = "directory", ResourceTypeEnum2.Document = "document", ResourceTypeEnum2.DocumentHistory = "document-history", ResourceTypeEnum2.Domain = "domain", ResourceTypeEnum2.EmailUuid = "email-uuid", ResourceTypeEnum2.Embed = "embed", ResourceTypeEnum2.Enterprise = "enterprise", ResourceTypeEnum2.Entitlement = "entitlement", ResourceTypeEnum2.EntitlementTemplate = "entitlement-template", ResourceTypeEnum2.EntityProperty = "entity-property", ResourceTypeEnum2.Environment = "environment", ResourceTypeEnum2.Escalation = "escalation", ResourceTypeEnum2.Event = "event", ResourceTypeEnum2.EventSource = "event-source", ResourceTypeEnum2.Extension = "extension", ResourceTypeEnum2.ExtensionGroup = "extension-group", ResourceTypeEnum2.Favourite = "favourite", ResourceTypeEnum2.FeatureFlag = "feature-flag", ResourceTypeEnum2.FeatureFlagHistory = "feature-flag-history", ResourceTypeEnum2.Feedback = "feedback", ResourceTypeEnum2.File = "file", ResourceTypeEnum2.Filter = "filter", ResourceTypeEnum2.FilterEmailSubscription = "filter-email-subscription", ResourceTypeEnum2.FlowEditor = "flow-editor", ResourceTypeEnum2.Form = "form", ResourceTypeEnum2.Goal = "goal", ResourceTypeEnum2.Group = "group", ResourceTypeEnum2.HelpCenter = "help-center", ResourceTypeEnum2.HelpDesk = "help-desk", ResourceTypeEnum2.HelpPointer = "help-pointer", ResourceTypeEnum2.Id = "id", ResourceTypeEnum2.ImportConfiguration = "import-configuration", ResourceTypeEnum2.Incident = "incident", ResourceTypeEnum2.IncidentAlertLink = "incident-alert-link", ResourceTypeEnum2.IncidentHistory = "incident-history", ResourceTypeEnum2.IncidentStatusUpdate = "incident-status-update", ResourceTypeEnum2.IncomingCallHistory = "incoming-call-history", ResourceTypeEnum2.IncomingWebhook = "incoming-webhook", ResourceTypeEnum2.Initiative = "initiative", ResourceTypeEnum2.Instance = "instance", ResourceTypeEnum2.Integration = "integration", ResourceTypeEnum2.IntentProjection = "intent-projection", ResourceTypeEnum2.IntentQuestionProjection = "intent-question-projection", ResourceTypeEnum2.IntentRuleProjection = "intent-rule-projection", ResourceTypeEnum2.IntentTemplate = "intent-template", ResourceTypeEnum2.InvoiceGroup = "invoice-group", ResourceTypeEnum2.Issue = "issue", ResourceTypeEnum2.IssueAttachment = "issue-attachment", ResourceTypeEnum2.IssueComment = "issue-comment", ResourceTypeEnum2.IssueFieldMetadata = "issue-field-metadata", ResourceTypeEnum2.IssueFieldOption = "issue-field-option", ResourceTypeEnum2.Issuefieldvalue = "issuefieldvalue", ResourceTypeEnum2.IssueHistory = "issue-history", ResourceTypeEnum2.IssueLinkType = "issue-link-type", ResourceTypeEnum2.IssueSearchView = "issue-search-view", ResourceTypeEnum2.IssueStatus = "issue-status", ResourceTypeEnum2.IssueType = "issue-type", ResourceTypeEnum2.IssueTypeScheme = "issue-type-scheme", ResourceTypeEnum2.IssueTypeSchemeMapping = "issue-type-scheme-mapping", ResourceTypeEnum2.IssueWorklog = "issue-worklog", ResourceTypeEnum2.JiraProjectAndDevopsToolRelationship = "jira-project-and-devops-tool-relationship", ResourceTypeEnum2.JiraProjectAndDocumentationSpaceRelationship = "jira-project-and-documentation-space-relationship", ResourceTypeEnum2.JiraProjectAndOperationsTeamRelationship = "jira-project-and-operations-team-relationship", ResourceTypeEnum2.JiraProjectAndVcsRepositoryRelationship = "jira-project-and-vcs-repository-relationship", ResourceTypeEnum2.JiraProjectAndVcsRepositoryRelationshipTempMigration = "jira-project-and-vcs-repository-relationship-temp-migration", ResourceTypeEnum2.LakeConnection = "lake-connection", ResourceTypeEnum2.Layout = "layout", ResourceTypeEnum2.Learning = "learning", ResourceTypeEnum2.LifecycleResource = "lifecycle-resource", ResourceTypeEnum2.LifecycleResourcePackage = "lifecycle-resource-package", ResourceTypeEnum2.LifecycleResourcePackageType = "lifecycle-resource-package-type", ResourceTypeEnum2.List = "list", ResourceTypeEnum2.LongRunningTask = "long-running-task", ResourceTypeEnum2.MetricDefinition = "metric-definition", ResourceTypeEnum2.MetricSource = "metric-source", ResourceTypeEnum2.Namespace = "namespace", ResourceTypeEnum2.Note = "note", ResourceTypeEnum2.Notification = "notification", ResourceTypeEnum2.NotificationType = "notification-type", ResourceTypeEnum2.NotificationTypeScheme = "notification-type-scheme", ResourceTypeEnum2.OauthClient = "oauthClient", ResourceTypeEnum2.OauthConsumer = "oauth-consumer", ResourceTypeEnum2.Object = "object", ResourceTypeEnum2.Offering = "offering", ResourceTypeEnum2.OfferingRelationshipTemplate = "offering-relationship-template", ResourceTypeEnum2.OperationsComponent = "operations-component", ResourceTypeEnum2.OperationsWorkspace = "operations-workspace", ResourceTypeEnum2.Order = "order", ResourceTypeEnum2.Org = "org", ResourceTypeEnum2.Organization = "organization", ResourceTypeEnum2.OrgUser = "org-user", ResourceTypeEnum2.Page = "page", ResourceTypeEnum2.PaymentMethod = "payment-method", ResourceTypeEnum2.Performance = "performance", ResourceTypeEnum2.PermissionScheme = "permission-scheme", ResourceTypeEnum2.Plan = "plan", ResourceTypeEnum2.Portal = "portal", ResourceTypeEnum2.PostIncidentReviewLink = "post-incident-review-link", ResourceTypeEnum2.Presentation = "presentation", ResourceTypeEnum2.PricingPlan = "pricing-plan", ResourceTypeEnum2.Priority = "priority", ResourceTypeEnum2.Product = "product", ResourceTypeEnum2.Program = "program", ResourceTypeEnum2.ProgramStatusUpdate = "program-status-update", ResourceTypeEnum2.Project = "project", ResourceTypeEnum2.ProjectCategory = "project-category", ResourceTypeEnum2.ProjectFeature = "project-feature", ResourceTypeEnum2.ProjectOverview = "project-overview", ResourceTypeEnum2.ProjectRoleActor = "project-role-actor", ResourceTypeEnum2.ProjectShortcut = "project-shortcut", ResourceTypeEnum2.ProjectType = "project-type", ResourceTypeEnum2.Promotion = "promotion", ResourceTypeEnum2.Provider = "provider", ResourceTypeEnum2.Pullrequest = "pullrequest", ResourceTypeEnum2.PullRequest = "pull-request", ResourceTypeEnum2.PullRequestHistory = "pull-request-history", ResourceTypeEnum2.Question = "question", ResourceTypeEnum2.Queue = "queue", ResourceTypeEnum2.Quote = "quote", ResourceTypeEnum2.Relationship = "relationship", ResourceTypeEnum2.RemoteLink = "remote-link", ResourceTypeEnum2.RemoteLinkHistory = "remote-link-history", ResourceTypeEnum2.Repository = "repository", ResourceTypeEnum2.RepositoryTransfer = "repository-transfer", ResourceTypeEnum2.RequestForm = "request-form", ResourceTypeEnum2.RequestType = "request-type", ResourceTypeEnum2.Resolution = "resolution", ResourceTypeEnum2.ResourceUsageMetric = "resource-usage-metric", ResourceTypeEnum2.ResourceUsageRecommendation = "resource-usage-recommendation", ResourceTypeEnum2.Role = "role", ResourceTypeEnum2.Rule = "rule", ResourceTypeEnum2.Schedule = "schedule", ResourceTypeEnum2.ScheduleRotation = "schedule-rotation", ResourceTypeEnum2.Schema = "schema", ResourceTypeEnum2.Scorecard = "scorecard", ResourceTypeEnum2.SecureTunnel = "secure-tunnel", ResourceTypeEnum2.SecurityContainer = "security-container", ResourceTypeEnum2.SecurityLevel = "security-level", ResourceTypeEnum2.SecurityWorkspace = "security-workspace", ResourceTypeEnum2.Service = "service", ResourceTypeEnum2.ServiceAndJiraProjectRelationship = "service-and-jira-project-relationship", ResourceTypeEnum2.ServiceAndOperationsTeamRelationship = "service-and-operations-team-relationship", ResourceTypeEnum2.ServiceAndVcsRepositoryRelationship = "service-and-vcs-repository-relationship", ResourceTypeEnum2.ServiceRelationship = "service-relationship", ResourceTypeEnum2.ShipToParty = "ship-to-party", ResourceTypeEnum2.Site = "site", ResourceTypeEnum2.SiteUser = "site-user", ResourceTypeEnum2.Sla = "sla", ResourceTypeEnum2.Space = "space", ResourceTypeEnum2.Spreadsheet = "spreadsheet", ResourceTypeEnum2.Sprint = "sprint", ResourceTypeEnum2.StreamhubSchema = "streamhub-schema", ResourceTypeEnum2.Subscription = "subscription", ResourceTypeEnum2.Tag = "tag", ResourceTypeEnum2.Task = "task", ResourceTypeEnum2.Team = "team", ResourceTypeEnum2.TeamCalendar = "team-calendar", ResourceTypeEnum2.TeamMember = "team-member", ResourceTypeEnum2.Timeline = "timeline", ResourceTypeEnum2.Tool = "tool", ResourceTypeEnum2.TransactionAccount = "transaction-account", ResourceTypeEnum2.Type = "type", ResourceTypeEnum2.User = "user", ResourceTypeEnum2.Userbase = "userbase", ResourceTypeEnum2.UserBoardConfig = "user-board-config", ResourceTypeEnum2.UserBroadcastMessage = "user-broadcast-message", ResourceTypeEnum2.UserDefinedParameter = "user-defined-parameter", ResourceTypeEnum2.UserGrant = "userGrant", ResourceTypeEnum2.UserProperty = "user-property", ResourceTypeEnum2.Version = "version", ResourceTypeEnum2.VersionApprover = "version-approver", ResourceTypeEnum2.ViewConfig = "view-config", ResourceTypeEnum2.Vulnerability = "vulnerability", ResourceTypeEnum2.VulnerabilityHistory = "vulnerability-history", ResourceTypeEnum2.Webhook = "webhook", ResourceTypeEnum2.Whiteboard = "whiteboard", ResourceTypeEnum2.WorkbenchResource = "workbench-resource", ResourceTypeEnum2.Workflow = "workflow", ResourceTypeEnum2.WorkflowScheme = "workflow-scheme", ResourceTypeEnum2.Worklog = "worklog", ResourceTypeEnum2.Workspace = "workspace", ResourceTypeEnum2))(ResourceTypeEnum || {});
export {
  AnalyticsUserAri,
  AnyAri,
  ApprovalDeskDecisionTableAri,
  ApprovalDeskRoleAri,
  ApprovalDeskWorkflowAri,
  ApprovalDeskWorkspaceAri,
  Ari,
  Arm,
  Ati,
  AtlashubIdAri,
  AtlashubSiteAri,
  AutomationRoleAri,
  AutomationRuleAri,
  AvocadoAnswerAri,
  AvocadoQuestionAri,
  AvocadoRoleAri,
  AvocadoSiteAri,
  AvocadoUserAri,
  AvpDashboardAri,
  AvpDatasourceAri,
  AvpRoleAri,
  AvpSiteAri,
  BeaconAlertAri,
  BeaconDetectionAri,
  BeaconRoleAri,
  BeaconSiteAri,
  BeaconSubscriptionAri,
  BeaconWorkspaceAri,
  BitbucketAppAri,
  BitbucketOauthConsumerAri,
  BitbucketProjectAri,
  BitbucketPullrequestAri,
  BitbucketRepositoryAri,
  BitbucketRepositoryTransferAri,
  BitbucketSiteAri,
  BitbucketTeamAri,
  BitbucketWorkspaceAri,
  BrieRoleAri,
  BrieSiteAri,
  CanvasBoardAri,
  CanvasDatabaseAri,
  CanvasSiteAri,
  CanvasWorkspaceAri,
  CmdbAttributeAri,
  CmdbImportConfigurationAri,
  CmdbObjectAri,
  CmdbSchemaAri,
  CmdbTypeAri,
  CmdbWorkspaceAri,
  CommerceCatalogAccountAri,
  CommerceChangeAri,
  CommerceEntitlementAri,
  CommerceEntitlementTemplateAri,
  CommerceInitiativeAri,
  CommerceInvoiceGroupAri,
  CommerceOfferingAri,
  CommerceOfferingRelationshipTemplateAri,
  CommerceOrderAri,
  CommercePaymentMethodAri,
  CommercePricingPlanAri,
  CommerceProductAri,
  CommercePromotionAri,
  CommerceQuoteAri,
  CommerceRoleAri,
  CommerceShipToPartyAri,
  CommerceTransactionAccountAri,
  CommerceWorkbenchResourceAri,
  CompassAssistantAnswerAri,
  CompassComponentAri,
  CompassComponentLabelAri,
  CompassComponentLinkAri,
  CompassCustomFieldDefinitionAri,
  CompassEventSourceAri,
  CompassIncomingWebhookAri,
  CompassMetricDefinitionAri,
  CompassMetricSourceAri,
  CompassRoleAri,
  CompassScorecardAri,
  CompassSiteAri,
  CompassUserDefinedParameterAri,
  CompassWebhookAri,
  CompassWorkspaceAri,
  ConfluenceAdminAnnouncementBannerAri,
  ConfluenceAttachmentAri,
  ConfluenceBlogpostAri,
  ConfluenceCommentAri,
  ConfluenceContentAri,
  ConfluenceDatabaseAri,
  ConfluenceEventAri,
  ConfluenceLongRunningTaskAri,
  ConfluenceNoteAri,
  ConfluencePageAri,
  ConfluenceRoleAri,
  ConfluenceSiteAri,
  ConfluenceSpaceAri,
  ConfluenceTaskAri,
  ConfluenceTeamCalendarAri,
  ConfluenceUserPropertyAri,
  ConfluenceWhiteboardAri,
  ConfluenceWorkspaceAri,
  ConversationalHelpConversationAri,
  ConversationalHelpConversationMessageAri,
  DataLakeLakeConnectionAri,
  DevopsContainerAri,
  DevopsNamespaceAri,
  DevopsProviderAri,
  DevopsRelationshipAri,
  DevopsToolAri,
  EcosystemAppAri,
  EcosystemAppEnvironmentVersionAri,
  EcosystemConnectAppAri,
  EcosystemEnvironmentAri,
  EcosystemExtensionAri,
  EcosystemExtensionGroupAri,
  ElementsCustomEmojiMetadataAri,
  ElementsWorkspaceAri,
  ExusExternalUserAri,
  FigmaFileAri,
  GoogleDocumentAri,
  GoogleFormAri,
  GooglePresentationAri,
  GoogleSpreadsheetAri,
  GraphBranchAri,
  GraphBranchHistoryAri,
  GraphBuildAri,
  GraphBuildHistoryAri,
  GraphCommitAri,
  GraphDeploymentAri,
  GraphDeploymentHistoryAri,
  GraphDesignAri,
  GraphDesignHistoryAri,
  GraphDocumentAri,
  GraphDocumentHistoryAri,
  GraphFeatureFlagAri,
  GraphFeatureFlagHistoryAri,
  GraphIncidentAri,
  GraphIncidentHistoryAri,
  GraphJiraProjectAndDevopsToolRelationshipAri,
  GraphJiraProjectAndDocumentationSpaceRelationshipAri,
  GraphJiraProjectAndOperationsTeamRelationshipAri,
  GraphJiraProjectAndVcsRepositoryRelationshipAri,
  GraphOperationsWorkspaceAri,
  GraphPullRequestAri,
  GraphPullRequestHistoryAri,
  GraphRemoteLinkAri,
  GraphRemoteLinkHistoryAri,
  GraphRepositoryAri,
  GraphSecurityContainerAri,
  GraphSecurityWorkspaceAri,
  GraphServiceAndJiraProjectRelationshipAri,
  GraphServiceAndOperationsTeamRelationshipAri,
  GraphServiceAndVcsRepositoryRelationshipAri,
  GraphServiceAri,
  GraphServiceRelationshipAri,
  GraphVulnerabilityAri,
  GraphVulnerabilityHistoryAri,
  GraphWorkspaceAri,
  HelpArticleAri,
  HelpChannelAri,
  HelpHelpCenterAri,
  HelpHelpDeskAri,
  HelpLayoutAri,
  HelpPortalAri,
  HelpRequestFormAri,
  IdentityAppAri,
  IdentityAuthPolicyAri,
  IdentityCustomerDirectoryAri,
  IdentityCustomerOrganizationAri,
  IdentityDirectoryAri,
  IdentityDomainAri,
  IdentityGroupAri,
  IdentityOauthClientAri,
  IdentityRoleAri,
  IdentitySiteAri,
  IdentityTeamAri,
  IdentityTeamMemberAri,
  IdentityUserAri,
  IdentityUserGrantAri,
  IdentityUserbaseAri,
  JiraAlignInstanceAri,
  JiraAnnouncementBannerAri,
  JiraAvailableFieldsAri,
  JiraBranchAri,
  JiraBranchHistoryAri,
  JiraBuildAri,
  JiraBuildHistoryAri,
  JiraBulkOperationTaskAri,
  JiraCommentAri,
  JiraCommitAri,
  JiraComponentAri,
  JiraConnectAppAri,
  JiraCoreRoleAri,
  JiraCoreSiteAri,
  JiraDashboardAri,
  JiraDeploymentAri,
  JiraDeploymentHistoryAri,
  JiraDesignAri,
  JiraDesignHistoryAri,
  JiraDocumentAri,
  JiraDocumentHistoryAri,
  JiraEntityPropertyAri,
  JiraFavouriteAri,
  JiraFeatureFlagAri,
  JiraFeatureFlagHistoryAri,
  JiraFilterAri,
  JiraFilterEmailSubscriptionAri,
  JiraIncidentAri,
  JiraIncidentHistoryAri,
  JiraIssueAri,
  JiraIssueAttachmentAri,
  JiraIssueCommentAri,
  JiraIssueFieldMetadataAri,
  JiraIssueFieldOptionAri,
  JiraIssueHistoryAri,
  JiraIssueLinkTypeAri,
  JiraIssueSearchViewAri,
  JiraIssueStatusAri,
  JiraIssueTypeAri,
  JiraIssueTypeSchemeAri,
  JiraIssueTypeSchemeMappingAri,
  JiraIssueWorklogAri,
  JiraIssuefieldvalueAri,
  JiraNotificationTypeAri,
  JiraNotificationTypeSchemeAri,
  JiraOperationsComponentAri,
  JiraOperationsWorkspaceAri,
  JiraPermissionSchemeAri,
  JiraPlanAri,
  JiraPostIncidentReviewLinkAri,
  JiraPriorityAri,
  JiraProjectAri,
  JiraProjectCategoryAri,
  JiraProjectFeatureAri,
  JiraProjectOverviewAri,
  JiraProjectRoleActorAri,
  JiraProjectShortcutAri,
  JiraProjectTypeAri,
  JiraPullRequestAri,
  JiraPullRequestHistoryAri,
  JiraRemoteLinkAri,
  JiraRemoteLinkHistoryAri,
  JiraRepositoryAri,
  JiraResolutionAri,
  JiraResourceUsageMetricAri,
  JiraResourceUsageRecommendationAri,
  JiraRoleAri,
  JiraSecurityContainerAri,
  JiraSecurityLevelAri,
  JiraSecurityWorkspaceAri,
  JiraServicedeskApprovalAri,
  JiraServicedeskCalendarEventAri,
  JiraServicedeskCannedResponseAri,
  JiraServicedeskOrganizationAri,
  JiraServicedeskQueueAri,
  JiraServicedeskRequestTypeAri,
  JiraServicedeskRoleAri,
  JiraServicedeskSiteAri,
  JiraServicedeskSlaAri,
  JiraSiteAri,
  JiraSoftwareBoardAri,
  JiraSoftwareBoardFeatureAri,
  JiraSoftwareBoardIssueListAri,
  JiraSoftwareCardAri,
  JiraSoftwareCardParentAri,
  JiraSoftwareColumnAri,
  JiraSoftwareCustomFilterAri,
  JiraSoftwareRoleAri,
  JiraSoftwareSiteAri,
  JiraSoftwareUserBoardConfigAri,
  JiraSprintAri,
  JiraUserBroadcastMessageAri,
  JiraVersionApproverAri,
  JiraVersionAri,
  JiraVulnerabilityAri,
  JiraVulnerabilityHistoryAri,
  JiraWorkflowAri,
  JiraWorkflowSchemeAri,
  JiraWorklogAri,
  JiraWorkspaceAri,
  LinkingPlatformDatasourceAri,
  MakerSpaceGroupAri,
  MarketingCustomerDomainAri,
  MeasurementEmailUuidAri,
  MeasurementSiteUserAri,
  MeasurementUserAri,
  MediaFileAri,
  MercuryCommentAri,
  MercuryProgramAri,
  MercuryProgramStatusUpdateAri,
  MercuryRoleAri,
  MercurySiteAri,
  OauthClientAri,
  OpsgenieAccountLoginAri,
  OpsgenieAccountSettingsAri,
  OpsgenieAlertAri,
  OpsgenieAlertRecipientLinkAri,
  OpsgenieApiRequestMetricAri,
  OpsgenieCallRoutingAri,
  OpsgenieCustomRoleAri,
  OpsgenieDeploymentAri,
  OpsgenieDeploymentServiceLinkAri,
  OpsgenieEscalationAri,
  OpsgenieEventAri,
  OpsgenieIncidentAlertLinkAri,
  OpsgenieIncidentAri,
  OpsgenieIncidentStatusUpdateAri,
  OpsgenieIncomingCallHistoryAri,
  OpsgenieIntegrationAri,
  OpsgenieNotificationAri,
  OpsgenieRoleAri,
  OpsgenieScheduleAri,
  OpsgenieScheduleRotationAri,
  OpsgenieSiteAri,
  OpsgenieTeamAri,
  OpsgenieTimelineAri,
  OpsgenieWorkspaceAri,
  PapiApiAri,
  PapiRoleAri,
  PassionfruitUserAri,
  PeoplePerftoolCalibrationAri,
  PeoplePerftoolFeedbackAri,
  PeoplePerftoolPerformanceAri,
  PlatformClassificationTagAri,
  PlatformLifecycleResourceAri,
  PlatformLifecycleResourcePackageAri,
  PlatformLifecycleResourcePackageTypeAri,
  PlatformOrgAri,
  PlatformOrgUserAri,
  PlatformSecureTunnelAri,
  PlatformServicesStreamhubSchemaAri,
  PlatformSiteAri,
  ResourceOwnerEnum as ResourceOwner,
  ResourceTypeEnum as ResourceType,
  RuntimeAuthClientRoleAri,
  SearchEventAri,
  StatuspageRoleAri,
  StatuspageSiteAri,
  SupportCustomerAri,
  TeamsTeamAri,
  TownsquareCommentAri,
  TownsquareGoalAri,
  TownsquareHelpPointerAri,
  TownsquareLearningAri,
  TownsquareProjectAri,
  TownsquareQuestionAri,
  TownsquareRoleAri,
  TownsquareSiteAri,
  TownsquareTagAri,
  TrelloBoardAri,
  TrelloCardAri,
  TrelloEnterpriseAri,
  TrelloListAri,
  TrelloRoleAri,
  TrelloSiteAri,
  TrelloUserAri,
  TrelloWorkspaceAri,
  UnifiedHelpRoleAri,
  UnifiedHelpSiteAri,
  UnifiedHelpWorkspaceAri,
  ValidationError,
  VirtualAgentConfigurationAri,
  VirtualAgentFlowEditorAri,
  VirtualAgentIntentProjectionAri,
  VirtualAgentIntentQuestionProjectionAri,
  VirtualAgentIntentRuleProjectionAri,
  VirtualAgentIntentTemplateAri
};
