// @atlaskit/onboarding/dist/es2019/components/SpotlightTransition.js
import React, { createContext } from 'react';
const TransitionContext = createContext({
    isOpen: true,
    onExited: () => { },
});
// checks if children exist and are truthy
const hasChildren = (children) => React.Children.count(children) > 0 &&
    React.Children.map(children, child => !!child).filter(Boolean).length > 0;
class Transition extends React.Component {
    constructor() {
        super(...arguments);
        this.state = {
            currentChildren: undefined,
        };
        this.onExited = () => {
            this.setState({
                currentChildren: this.props.children,
            });
        };
    }
    static getDerivedStateFromProps(props, state) {
        const { currentChildren: previousChildren } = state;
        const exiting = hasChildren(previousChildren) && !hasChildren(props.children);
        return {
            currentChildren: exiting ? previousChildren : props.children,
        };
    }
    render() {
        return (React.createElement(TransitionContext.Provider, { value: {
                onExited: this.onExited,
                isOpen: hasChildren(this.props.children),
            } }, this.state.currentChildren));
    }
}
export const SpotlightTransitionConsumer = TransitionContext.Consumer;
export default Transition;
