// @atlaskit/avatar/dist/es2019/AvatarItem.js
/** @jsx jsx */
// eslint-disable-next-line @repo/internal/fs/filename-pattern-match
import { createElement, forwardRef, Fragment } from 'react';
import { ClassNames, css, jsx } from '@emotion/react';
import { B200, B50, N30 } from '@atlaskit/theme/colors';
import { borderRadius } from '@atlaskit/theme/constants';
import { BORDER_WIDTH } from './constants';
import Text from './Text';
import { getButtonProps, getCustomElement, getLinkProps } from './utilities';
const avatarItemStyles = css({
  minWidth: 0,
  maxWidth: '100%',
  flex: '1 1 100%',
  lineHeight: '1.4',
  paddingInlineStart: "var(--ds-space-100, 8px)"
});
const getStyles = (css, {
  backgroundColor,
  isInteractive,
  isDisabled
}) =>
// eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage, @repo/internal/react/no-css-string-literals
css`
    align-items: center;
    background-color: ${backgroundColor};
    border-radius: ${borderRadius()}px;
    border: ${BORDER_WIDTH}px solid transparent;
    box-sizing: border-box;
    color: inherit;
    display: flex;
    font-size: inherit;
    font-style: normal;
    font-weight: normal;
    line-height: 1;
    outline: none;
    margin: ${"var(--ds-space-0, 0px)"};
    padding: ${"var(--ds-space-050, 4px)"};
    text-align: left;
    text-decoration: none;
    width: 100%;

    ${isInteractive && `
        :hover {
          background-color: ${`var(--ds-background-neutral-subtle-hovered, ${N30})`};
          cursor: pointer;
          text-decoration: none;
        }

        :focus {
          outline: none;
          border-color: ${`var(--ds-border-focused, ${B200})`};
        }

        :active {
          background-color: ${`var(--ds-background-neutral-subtle-pressed, ${B50})`};
        }
      `}

    ${isDisabled && `
        cursor: not-allowed;
        opacity: ${"var(--ds-opacity-disabled, 0.5)"};
        pointer-events: none;
      `}
  `;

/**
 * __Avatar item__
 *
 * An avatar item is a wrapper that goes around an avatar when it's displayed alongside text, such as a name or status.
 *
 * - [Examples](https://atlassian.design/components/avatar/avatar-item/examples)
 * - [Code](https://atlassian.design/components/avatar/avatar-item/code)
 */
const AvatarItem = /*#__PURE__*/forwardRef(({
  avatar,
  backgroundColor = 'transparent',
  children,
  isTruncationDisabled,
  href,
  isDisabled,
  onClick,
  primaryText,
  secondaryText,
  target,
  testId,
  label
}, ref) => {
  const getTestId = (testId, children) => !children ? {
    'data-testid': `${testId}--itemInner`
  } : {
    testId: `${testId}--itemInner`
  };
  const componentProps = () => {
    if (isDisabled) {
      return {
        disabled: true
      };
    }

    // return only relevant props for either anchor or button elements
    return {
      ...(href && getLinkProps(href, target)),
      ...(onClick && !href ? getButtonProps(onClick) : {
        onClick
      })
    };
  };
  return jsx(ClassNames, null, ({
    css
  }) => {
    const props = {
      ref,
      className: getStyles(css, {
        backgroundColor,
        isInteractive: Boolean(onClick || href),
        isDisabled
      }),
      ...componentProps(),
      ...(testId && getTestId(testId, children)),
      ...((onClick || href) && {
        'aria-label': label
      }),
      children: jsx(Fragment, null, avatar, jsx("div", {
        css: avatarItemStyles
      }, jsx(Text, {
        shouldTruncate: !isTruncationDisabled
      }, primaryText), jsx(Text, {
        isSecondary: true,
        shouldTruncate: !isTruncationDisabled
      }, secondaryText)))
    };
    return children ? children(props) : /*#__PURE__*/createElement(getCustomElement(isDisabled, href, onClick), props);
  });
});
AvatarItem.displayName = 'AvatarItem';
export default AvatarItem;
