// @atlaskit/avatar/dist/cjs/Text.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
/** @jsx jsx */
// eslint-disable-next-line @repo/internal/fs/filename-pattern-match

var baseStyles = (0, _react.css)({
  display: 'block',
  margin: "var(--ds-space-0, 0px)",
  color: "var(--ds-text, ".concat(_colors.N900, ")")
});
var truncateStyles = (0, _react.css)({
  overflowX: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
});
var secondaryStyles = (0, _react.css)({
  color: "var(--ds-text-subtlest, ".concat(_colors.N200, ")"),
  fontSize: '0.85em'
});

/**
 * __Text__
 *
 * Text is used internally only.
 */
var Text = function Text(_ref) {
  var isSecondary = _ref.isSecondary,
    children = _ref.children,
    shouldTruncate = _ref.shouldTruncate;
  return (0, _react.jsx)("span", {
    css: [baseStyles, isSecondary && secondaryStyles, shouldTruncate && truncateStyles]
  }, children);
};
var _default = exports.default = Text;
